### DISCUSSION TITLE
Multiple async var assignment

### DISCUSSION THREAD

**Discussion by israellot** [dotnet/csharplang#5524] [Language Ideas] [answered]
Firing multiple async tasks and assigning results to variables in parallel could be simplified

```
var catTask = FeedCat();
var houseTask = SellHouse();
var carTask = BuyCar();

await Task.WhenAll(catTask, houseTask, carTask);

var cat = await catTask;
var house = await houseTask;
var car = await carTask;
```

This could be a one liner :

`var (cat,house,car) = await (FeedCat(),SellHouse(), BuyCar())`

**Comment by huoyaoyuan** [comment, marked as answer]
The semantics is unclear. Are they done sequentially or parallelly? Which `ConfigureAwait` option is used?
This can be done with an extension method on tuple of tasks. However, the behavior needs to be clarified first.

**Reply by huoyaoyuan** [reply]
See also #1454

**Reply by svick** [reply]
There's also https://github.com/dotnet/runtime/issues/20166, where the conclusion was that you should use the NuGet package [TaskTupleAwaiter](https://www.nuget.org/packages/TaskTupleAwaiter/) for this.

I am marking this as duplicate of https://github.com/dotnet/csharplang/discussions/1454.

**Reply by jnm2** [reply]
@huoyaoyuan I don't think the semantics are unclear once you realize how easy it is to write a series of awaits. ConfigureAwait is easy, too. TaskTupleAwaiter has `.ConfigureAwait` on tuples of tasks. Whether or not ConfigureAwait is specified on the tuple, it desugars in a straightforward way.
